web/src/peer_data.ts

```diff
@@ -29,23 +30,43 @@ const pending_subscriber_requests = new Map<
 export function clear_for_testing(): void {
     stream_subscribers.clear();
     fetched_stream_ids.clear();
+    pending_subscriber_requests.clear();
 }
 
 const fetch_stream_subscribers_response_schema = z.object({
     subscribers: z.array(z.number()),
 });
 
+// This function will always resolve to a LazySet but could hang
+// indefinitely trying to fetch data. `num_attempts` should only
+// be set in the recursive call.
+export async function maybe_fetch_stream_subscribers_with_retry(
+    stream_id: number,
+    num_attempts = 1,
+): Promise<LazySet> {
+    const subscribers = await maybe_fetch_stream_subscribers(stream_id);
+    if (subscribers === null) {
+        num_attempts += 1;
+        // TODO: figure out how to handle sending xhr here
```

Comment:
This code feels cleaner except that I'm not sure what to send for the `xhr` parameter of `get_retry_backoff_seconds`. Would it be reasonable to make that an optional parameter?

Reply:
Yeah that parameter can be optional.